src/bots/filler.ts

```diff
@@ -2310,169 +2324,220 @@ export class FillerBot implements Bot {
+										settlePnlPromises.push(
+											this.driftClient.txSender.sendVersionedTransaction(
+												simResult.tx,
+												[],
+												this.driftClient.opts,
+												true
+											)
+										);
+									}
+								} else {
+									logger.info(`dry run, skipping settlePnls)`);
 								}
-							} else {
-								logger.info(`dry run, skipping settlePnls)`);
 							}
+						} catch (err) {
+							if (!(err instanceof Error)) {
+								return;
+							}
+							const errorCode = getErrorCode(err) ?? 0;
+							logger.error(
+								`Error code: ${errorCode} while settling pnls for markets ${JSON.stringify(
+									marketIds
+								)}: ${err.message}`
+							);
+							console.error(err);
 						}
-					} catch (err) {
-						if (!(err instanceof Error)) {
-							return;
-						}
-						const errorCode = getErrorCode(err) ?? 0;
-						logger.error(
-							`Error code: ${errorCode} while settling pnls for markets ${JSON.stringify(
-								marketIds
-							)}: ${err.message}`
-						);
-						console.error(err);
 					}
-				}
-				try {
-					const txs = await Promise.all(settlePnlPromises);
-					for (const tx of txs) {
-						logger.info(
-							`Settle positive PNLs tx: https://solscan/io/tx/${tx.txSig}`
-						);
+					try {
+						const txs = await Promise.all(settlePnlPromises);
+						for (const tx of txs) {
+							logger.info(
+								`Settle positive PNLs tx: https://solscan/io/tx/${tx.txSig}`
+							);
+						}
+					} catch (e) {
+						logger.error(`Error settling positive pnls: ${e}`);
 					}
-				} catch (e) {
-					logger.error(`Error settling positive pnls: ${e}`);
+					this.lastSettlePnl = now;
 				}
-				this.lastSettlePnl = now;
 			}
-		}
-		if (this.rebalanceFiller) {
-			logger.info(`Rebalancing filler`);
-			const fillerSolBalance = await this.driftClient.connection.getBalance(
-				this.driftClient.authority
-			);
-			this.hasEnoughSolToFill = fillerSolBalance >= this.minimumAmountToFill;
-			if (!this.hasEnoughSolToFill && this.jupiterClient !== undefined) {
-				logger.info(`Swapping USDC for SOL to rebalance filler`);
-				swapFillerHardEarnedUSDCForSOL(
-					this.priorityFeeSubscriber,
-					this.driftClient,
-					this.jupiterClient,
-					await this.getBlockhashForTx()
-				).then(async () => {
-					const fillerSolBalanceAfterSwap =
-						await this.driftClient.connection.getBalance(
-							this.driftClient.authority,
-							'processed'
-						);
-					this.hasEnoughSolToFill =
-						fillerSolBalanceAfterSwap >= this.minimumAmountToFill;
-				});
-			} else {
-				this.hasEnoughSolToFill = true;
+
+			if (isTotalUnsettledPnlSettlable) {
+				await this.rebalance();
 			}
 		}
 	}
 
+	protected async rebalance() {
+		logger.info(`Rebalancing filler`);
+		const fillerSolBalance = await this.driftClient.connection.getBalance(
+			this.driftClient.authority
+		);
+
+		this.hasEnoughSolToFill = fillerSolBalance >= this.minimumAmountToFill;
```

Comment:
small nit but we probably want this to be set outside of rebalance -- that way even if someone toggles off auto-rebalancing, we still set the flag if it drops below minAmountToFill so they aren't constantly simulating txs that fail bc of not enough SOL in wallet